code/game/machinery/hivebot_swarm_core.dm

```diff
@@ -0,0 +1,111 @@
+/obj/machinery/hivebot_swarm_core //A hulking machine armed with several weapons. Creates hivebots and defends itself with a buzzsaw and mounted weapons.
+	name = "hivebot swarm core"
+	desc = "An incredibly dangerous machine with random weapons haphazardly attached with spot welds. Its main body seems impervious to damage, but the head is vulnerable."
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "hivebot_swarm_core"
+	pixel_x = -32
+	pixel_y = -16
+	use_power = FALSE
+	obj_integrity = 250
+	max_integrity = 250
+	anchored = TRUE
+	density = TRUE
+	var/recharge_period = 0 //The number of ticks the core is waiting for and cannot do anything
+	var/mob/living/threat_to_swarm //The core's "target" for its attacks
+	var/hivebot_limit = 40 //Stop making hivebots if there are this many in a 7x7 square around us
+
+/obj/machinery/hivebot_swarm_core/Destroy()
+	visible_message("<span class='warning'>[src] falls apart, its red eyes sputtering out.</span>")
+	playsound(src, 'sound/magic/clockwork/anima_fragment_death.ogg', 50, 0)
+	new/obj/structure/fluff/hivebot_swarm_core(get_turf(src))
+	return ..()
+
+/obj/machinery/hivebot_swarm_core/examine(mob/user)
+	..()
+	if(recharge_period)
+		to_chat(user, "<span class='warning'>It seems to be recharging.</span>")
+
+/obj/machinery/hivebot_swarm_core/process()
+	if(recharge_period)
```

Comment:
It only dispenses if there's something nearby, that's why it processes along with the cooldown